services/app/src/components/views/Search.test.tsx

```diff
@@ -56,4 +74,39 @@ describe('Search', () => {
     const {e} = setup({user: testLoggedInUser, searching: true});
     expect(e.find('.lds-ellipsis').length).toEqual(1);
   });
+  test('searches if user is logged in and no results', (done) => {
+    const {props, e} = setupMount({
+      user: {
+        loggedIn: true,
+      },
+      results: null,
+    });
+    setTimeout(() => {
+      expect(props.onSearch).toHaveBeenCalled();
+      done();
+    }, 100);
```

Comment:
Fighting timeouts with timeouts is a pretty huge anti-pattern in tests (easy to have race conditions).

What if instead we moved the async `setTimeout` logic to the container, so we can easily detect whether onSearch is called synchronously or not?